vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
```

Comment:
>JsiDataHandler [](start = 17, length = 14)

Do we need to have information about encoding here? Lower in this file you mention getting it from various locations, which can have various encodings... If there is a reuiqred encoding for JSI (probably UTF8) we should add a comment ot make sure the byte aray is encoed that way.

Reply:
The IJsiBuffer is just a byte buffer. It can be used for binary data. It does not have any encoding associated with it. The encoding only appears when we use it in some specific APIs. These APIs add the semantic for the byte buffer. I will change names to the IJsiByteBuffer and JsiByteBufferHanlder to make it more explicit.